### PR TITLE
refactor: Trajectories stores MultiTrajectory pointer

### PR DESCRIPTION
**Pull request by paulgessinger**
This is needed to be able to write a general tracks to trajectories converter.

This requires writing the tracks to the white board so the track state container stays alive after the algorithm ends.

Blocked by:
- #1836 
- #1837 
- #1838

### CODE REVIEW

File: Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/TrackFindingAlgorithm.hpp [hunk 1]
```diff
@@ -68,6 +68,7 @@ class TrackFindingAlgorithm final : public BareAlgorithm {
     std::string inputInitialTrackParameters;
     /// Output find trajectories collection.
     std::string outputTrajectories;
+    std::string outputTracks;
```
Comment (andiwand) [inline Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/TrackFindingAlgorithm.hpp:71]:
```suggestion
    /// Output find tracks collection.
    std::string outputTracks;
```

File: Examples/Algorithms/TrackFittingChi2/include/ActsExamples/TrackFittingChi2/TrackFittingChi2Algorithm.hpp [hunk 2]
```diff
@@ -60,6 +60,7 @@ class TrackFittingChi2Algorithm final : public BareAlgorithm {
     std::string inputInitialTrackParameters;
     /// Output fitted trajectories collection.
     std::string outputTrajectories;
+    std::string outputTracks;
```
Comment (andiwand) [inline Examples/Algorithms/TrackFittingChi2/include/ActsExamples/TrackFittingChi2/TrackFittingChi2Algorithm.hpp:63]:
```suggestion
    /// Output fitted tracks collection.
    std::string outputTracks;
```

**Review by andiwand — APPROVED** [review]
looks good to me :+1:

**Comment by paulgessinger** [comment]
Applied suggestions, can you reapprove @andiwand?

**Review by andiwand — APPROVED** [review]

**Pull request merged**